sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -235,6 +237,14 @@ private void drainQueue() {
             if (requestedMessages != Long.MAX_VALUE) {
                 numberRequested = REQUESTED.addAndGet(this, -numberConsumed);
             }
+
+            // In RECEIVE_AND_DELETE, we cannot just release message when there is no downstream and prefetch is
+            // disabled, because the message is deleted in service side, hence if `numberConsumed` is not increased and
+            // no `currentDownstream`, stop `drainQueue()` until next message or queue work is added to invoke `drainQueue()`.
+            currentDownstream = getOrUpdateCurrentWork();
+            if (numberConsumed == 0 && currentDownstream == null) {
+                break;
+            }
```

Comment:
Do we really need this check to exit the drain loop? In the next iteration of the drain loop, we exit if there is no current work

Reply:
there is another loop,
```java
while (numberRequested != 0L && !isEmpty) {
```
we have no current work, the requested != 0 and the buffer queue is not empty, this loop will never exit.

Reply:
there are three loop in this method:
```java
while (numberRequested != numberConsumed) {
```
this loop will exit when re-buffer messages or numberConsumed equal requested.

```java
while (!isEmitted) {
```
this loop will exit when emit the message or no current work.

```java
while (numberRequested != 0L && !isEmpty) {
```
this loop I mentioned above, in that scenario, it will never exit unless add a new queue work.

Reply:
this means in early code, if there is no current queue work, requested is not 0 and buffer message queue is not empty, the loop cannot exit, it will be hanging there.

Reply:
if we have some messages to release and just check `currentDownstream == null`, these message will remain in buffer until next work come in.

Reply:
try also avoid busy loop (that there is no downstream, but loop continue executing without any wait on other resources, using CPU)

Reply:
Kun suggest move this check into check current downtream,  that's more clear.